### ISSUE TITLE
Result counter and timer don't update when map gets populated on Chrome

### ISSUE THREAD

**Issue by KasperZutterman** [comunica/jQuery-Widget.js#90]
While the query is running the counter and clock are updating, but when the map is shown these fields freeze. Only when I press Stop Execution it gets updated.

[Sample Query](https://kadaster.github.io/datascience-comunica-client/#datasources=sparql%40https%3A%2F%2Fapi.labs.kadaster.nl%2Fdatasets%2Fkadaster%2Fkg%2Fservices%2Fdefault%2Fsparql&query=query%20%7B%0A%20%20id%0A%20%20naam%0A%20%20type(_%3Aschool)%0A%20%20locatie%20%7B%0A%20%20%20%20naam%0A%20%20%7D%0A%20%20geometry%0A%20%20geometry%20%7B%0A%20%20%20%20wkt%0A%20%20%7D%20%0A%7D&queryContext=%7B%0A%20%20%22%40context%22%3A%20%7B%0A%20%20%20%20%22geometry%22%3A%20%22https%3A%2F%2Fschema.org%2Fgeo%22%2C%0A%20%20%20%20%22wkt%22%3A%20%22https%3A%2F%2Fschema.org%2Fpolygon%22%2C%0A%20%20%20%20%22type%22%3A%20%22https%3A%2F%2Fschema.org%2FadditionalType%22%2C%0A%20%20%20%20%22school%22%3A%20%22https%3A%2F%2Fbrt.basisregistraties.overheid.nl%2Fbrt%2Fid%2FtypeGebouw%2Fschool%22%2C%0A%20%20%20%20%22naam%22%3A%20%22https%3A%2F%2Fschema.org%2Fname%22%2C%0A%20%20%20%20%22locatie%22%3A%20%22https%3A%2F%2Fschema.org%2FcontainedInPlace%22%0A%20%20%7D%0A%7D&resultsToTree=false&queryFormat=graphql)

![image](https://user-images.githubusercontent.com/21260838/123796647-a1e23900-d8e5-11eb-99e2-902f5328cd9d.png)

This problem occurs on Google Chrome, on Firefox this isn't an issue.

**Comment by rubensworks** [comment]
For reference, this is probably because the code doesn't allow the render loop to take place.

Could by fixed by either making the map logic faster, letting it run in a separate worker (not sure if that would work), or only adding points to the map every X ticks.

**Comment by pietercolpaert** [comment]
Indeed: it seems that now also the map will recalculate how it should be centered on each point that gets added, which requires also to check which tiles need to be downloaded on which zoom level. Probably buffering could be of help here?

**Comment by rubensworks** [comment]
A bounty has been placed on this issue via the Comunica Association (see original post).

**Comment by milafrerichs** [comment]
> As discussed with @rubensworks, I will work on this issue.

**Comment by milafrerichs** [comment]
I tested it with the query below in my Chrome and it seems to be working fine. 

Regardless we can/should add a way to make this less blocking. 
One method would be to use `requestAnimationFrame` for adding the markers. 
We could wrap the content of `handleGeospatialResult` (https://github.com/comunica/jQuery-Widget.js/blob/master/src/ldf-client-ui.js#L854) in a function and call it with `requestAnimationFrame` 

another alternative would be to use something like [MarkerCluster](https://github.com/Leaflet/Leaflet.markercluster) (a leaflet plugin). 

we could also batch the results by adding them to an array and after x results we add the batch to the map. 

what are your thoughts @rubensworks  

### Query:
```sparql
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>

SELECT DISTINCT * WHERE {
  ?item wdt:P31/wdt:P279* wd:Q16917;
        wdt:P625 ?geo .
}
```

**Comment by rubensworks** [comment]
@milafrerichs Your `requestAnimationFrame` proposal seems like the right way forward indeed.
The map results in `_addResult` should probably be buffered, and then flushed on each `requestAnimationFrame` call.

MarkerCluster could (in addition) also be a solution, but that doesn't seem required for this issue (but feel free to submit a separate PR for it if you'd like that).
